src/replication.c

```diff
@@ -1568,14 +1568,17 @@ void rdbPipeReadHandler(struct aeEventLoop *eventLoop, int fd, void *clientData,
 
         if (stillAlive == 0) {
             serverLog(LL_WARNING,"Diskless rdb transfer, last replica dropped, killing fork child.");
+            aeDeleteFileEvent(server.el, server.rdb_pipe_read, AE_READABLE);
             killRDBChild();
         }
-        /*  Remove the pipe read handler if at least one write handler was set. */
-        if (server.rdb_pipe_numconns_writing || stillAlive == 0) {
-            if (server.rdb_pipe_read != -1)
-                aeDeleteFileEvent(server.el, server.rdb_pipe_read, AE_READABLE);
+        /* Remove the pipe read handler if at least one write handler was set,
+         * or if we are restarting a BGSAVE for other replicas. */
+        else if (server.rdb_pipe_numconns_writing) {
+            aeDeleteFileEvent(server.el, server.rdb_pipe_read, AE_READABLE);
             break;
         }
+
+
```

Comment:
unwanted edit?

Reply:
because there is a problem with `aeDeleteFileEvent` before `killRDBChild`, it will cause repeated bgsave, I haven't had time to deal with it, please wait for me to ping you.

Reply:
not sure how this is related to the two empty lines..

Reply:
it's unnecessary, i haven't had time to commit a fix for it.  😅